ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/internalresponse/EthAccountsResultProvider.java

```diff
@@ -45,7 +48,28 @@ public EthAccountsResultProvider(final Supplier<Set<String>> addressesSupplier)
       throw new JsonRpcException(JsonRpcError.INVALID_PARAMS);
     }
 
-    return addressesSupplier.get().stream().sorted().collect(Collectors.toList());
+    final List<String> allAddress =
+        transactionSignerProvider.availableAddresses().stream()
+            .sorted()
```

Comment:
you can probably use filter on stream rather than re-iterating again to return availableAddress.

Reply:
I really wanted to - but couldn't work it out :/ Will give it another shot :)